### PR TITLE
Odometry Docs

### PR DESCRIPTION
**Pull request by theol0403**
### Description of the Change

This PR adds an odometry walkthrough tutorial, as well as various documentation improvements.

The odometry walkthrough is not yet completed, but I am submitting what I have so far so that the review process can get started. I will have time to finish the tutorial tomorrow night.

**Changes**:

- Remove unnecessary period from title - it was causing double period in footer.
- Remove all `using namespace okapi` from examples - it did not add value to the examples and users are expected to include the using namespace from `main.h` anyways.
- Add note about following motion profile paths backwards
- Format motion profile examples
- Add odometry walkthrough
- Improve main page
- Remove `LogLevel::warn` link in logger tutorial to encourage clicking on the link to `LogLevel`.

### Motivation

This will help users understand how to use odometry.

### CODE REVIEW

**Comment by Octogonapus** [comment]
> Remove all using namespace okapi from examples - it did not add value to the examples and users are expected to include the using namespace from main.h anyways.

I kinda disagree with this. We see users in Discord not know to include the namespace, so I put it in the examples to make it clear that it uses the namespace.

File: docs/tutorials/concepts/logging.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 ## Enable the Default Logger
 
 OkapiLib has a [Logger](@ref okapi::Logger) class which is used internally by many of
-OkapiLib's classes. It is a good idea to use the [warn](@ref okapi::Logger::LogLevel::warn)
+OkapiLib's classes. It is a good idea to use the warn
```
Comment (Octogonapus) [inline docs/tutorials/concepts/logging.md:6]: What good does this do?

File: docs/tutorials/walkthrough/chassisControllerBuilder.md [hunk 2]
```diff
@@ -148,6 +148,8 @@ ChassisControllerBuilder()
     .buildOdometry()
 ```
 
+For more information about odometry, read the [Odometry Tutorial](docs/tutorials/walkthrough/odometry.md).
```
Comment (Octogonapus) [inline docs/tutorials/walkthrough/chassisControllerBuilder.md]: Maybe also link to this under `## Configuring odometry`?

File: docs/tutorials/walkthrough/odometry.md [hunk 3]
```diff
@@ -0,0 +1,163 @@
+# Odometry
+
+Odometry is the act of tracking a robot's absolute position. 
+This information can be used by motion algorithms to drive to positions or turn to absolute angles.
+
+An excellent overview of the odometry tracking algorithm can be found [here](https://www.vexforum.com/t/team-5225-introduction-to-position-tracking-document/49640).
+
+# Configuring Odometry
```
Comment (Octogonapus) [inline docs/tutorials/walkthrough/odometry.md]:
```suggestion
## Configuring Odometry
```

File: docs/tutorials/walkthrough/odometry.md [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+OkapiLib supports odometry for all chassis configurations, and is configured using
```
Comment (Octogonapus) [inline docs/tutorials/walkthrough/odometry.md]: Can you rephrase this? It sounds awkward.

File: docs/tutorials/walkthrough/odometry.md [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+        {'E', 'F'}  // middle encoder in ADI ports E & F
+    )
+    // green gearset, 4 inch wheel diameter, 11 inch wheelbase
+    // 1 inch middle encoder distance, and 2.75 inch middle wheel diameter
+    .withDimensions(AbstractMotor::gearset::green, {{4_in, 11_in, 1_in, 2.75_in}, imev5GreenTPR})
+    .withOdometry() // use the same scales as the chassis (above)
+    .buildOdometry() // build an odometry chassis
+```
+
+# Using Odometry
```
Comment (Octogonapus) [inline docs/tutorials/walkthrough/odometry.md]:
```suggestion
## Using Odometry
```

**Review by Octogonapus — CHANGES_REQUESTED** [review]

**Comment by theol0403** [comment]
> I kinda disagree with this. We see users in Discord not know to include the namespace, so I put it in the examples to make it clear that it uses the namespace.

It's up to you to make the final call on this, but I think that more users will read the getting started guide with the new docs. I just think it directs the user's attention to the more important parts of the example, and that it is not necessary in most user's setups.

File: docs/tutorials/concepts/logging.md (hunk 1 above)
Comment (theol0403) [reply docs/tutorials/concepts/logging.md:6]: Here is how it looked before: 
image

With the line wrap the way it is, it makes it seem like the two words are part of the same link, which can be confusing.

When the user clicks on **warn**, it jumps to the anchor so that the rest of the enum is hidden:
image

In my opinion, it is more user friendly when the user clicks on **LogLevel**, because then it puts the entire enum into view:

image


Only having one link will make it so the user will be navigated to where all the modes are shown, instead of just the warn enum, which has no additional information.

File: docs/tutorials/walkthrough/odometry.md (hunk 4 above)
Comment (theol0403) [reply docs/tutorials/walkthrough/odometry.md]:
```suggestion
OkapiLib supports odometry for all chassis configurations. A chassis controller that uses odometry is configured using the 
```
Better?

**Comment by theol0403** [comment]
I'm done with this PR, once we decide what to do with `using namespace okapi`.
I think v4 is ready to be released after this is merged.

In the future, a concept tutorial about the use of `auto` in the examples and how to forward declare across multiple files would be valuable.

**Review by Octogonapus — APPROVED** [review]

**Comment by Octogonapus** [comment]
> I think v4 is ready to be released after this is merged

Not quite ready because I need to setup something to show people that the v4 docs are in a new location. As far as the artifact and new docs are concerned, though, I agree they are as ready as we can get them.
